liota/lib/utilities/filters/windowed_filters/windowed_range_filter.py

```diff
@@ -0,0 +1,67 @@
+#                                                                             #
+#  The BSD 2-Clause License                                                   #
+#                                                                             #
+#  Redistribution and use in source and binary forms, with or without         #
+#  modification, are permitted provided that the following conditions are met:#
+#                                                                             #
+#  - Redistributions of source code must retain the above copyright notice,   #
+#      this list of conditions and the following disclaimer.                  #
+#                                                                             #
+#  - Redistributions in binary form must reproduce the above copyright        #
+#      notice, this list of conditions and the following disclaimer in the    #
+#      documentation and/or other materials provided with the distribution.   #
+#                                                                             #
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"#
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE  #
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE #
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE  #
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR        #
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF       #
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS   #
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN    #
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)    #
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF     #
+#  THE POSSIBILITY OF SUCH DAMAGE.                                            #
+# ----------------------------------------------------------------------------#
+
+import logging
+
+from liota.lib.utilities.filters.range_filter import RangeFilter
+from liota.lib.utilities.filters.windowed_filters.windowed_filter import WindowedFilter
+
+log = logging.getLogger(__name__)
+
+
+class WindowedRangeFilter(RangeFilter, WindowedFilter):
+    """
+    RangeFilter with windowing scheme.
+
+    It keeps track of a configurable time window.  Even if all values has been filtered out at the
+    end of every time window, collected value is returned so that DCC is aware of it.
+    """
+
+    def __init__(self, filter_type=None, lower_bound=None, upper_bound=None, window_size_sec=10):
+        """
+        :param filter_type: Any one type from Filter Enum of RangeFilter.
+        :param lower_bound: Lower bound
+        :param upper_bound: Upper bound
+        :param window_size_sec: Configurable time window for heartbeat
+        """
+        RangeFilter.__init__(self, filter_type, lower_bound, upper_bound)
```

Comment:
With base classes `__init()__` having different signature, it is not possible to use `super()` in derived class. [Reference](http://stackoverflow.com/questions/16855087/how-does-multiple-inheritance-work-with-the-super-and-different-init-arg)
- One approach is to design base classes for multiple inheritance. [Reference] (http://stackoverflow.com/questions/34884567/python-multiple-inheritance-passing-arguments-to-constructors-using-super) (Possible use of kwargs)
- Other way is to use `BaseClass.__init__()`

So, I thought of using this approach.  Your opinion on this please ?